robot/rospackages/src/mcu_control/scripts/NavigationNode.py

```diff
@@ -19,56 +19,69 @@ def subscriber_callback(message):
         rover['heading'] = message.z
         hasHeading = True
 
-    if gotGpsPos:
-        Rov_to_des_distance = Distance(rover['latitude'], rover['longitude'], \
-        gpsGoal['latitude'], gpsGoal['longitude'])
-        Rov_to_des_direction = Direction(rover['latitude'], rover['longitude'], \
-        gpsGoal['latitude'], gpsGoal['longitude'])
+    if rospy.get_param('has_gps_goal'):
+        rov_to_des_distance = Distance(rover['latitude'], rover['longitude'],
+                                       gpsGoal['latitude'], gpsGoal['longitude'])
+        rov_to_des_direction = Direction(rover['latitude'], rover['longitude'],
+                                         gpsGoal['latitude'], gpsGoal['longitude'])
 
         if hasHeading:
-            Direction_adjust = Turning(Rov_to_des_direction, rover['heading'])
+            direction_adjust = Turning(rov_to_des_direction, rover['heading'])
         else:
-            Direction_adjust = -999 # no heading, give invalid number
+            direction_adjust = -999  # no heading, give invalid number
 
         msg = Point()
         # note that direction is based on compass directions where E is 90 and W is -90
-        msg.x = Direction_adjust
-        msg.y = Rov_to_des_distance
+        msg.x = direction_adjust
+        msg.y = rov_to_des_distance
+        if(msg.y < 50):
+            rospy.loginfo('moving on to next coorinates')
```

Comment:
hmm why 50? Is this log useful

Also I would recommend to use actual ros objects instead of x and y since now we know it works just fine :)

Reply:
*coordinates

Reply:
@MartensCedric I'm not sure, im not the one who coded this part. Im assuming that once the distance to the goal is less than 50 meters, the GUI automatically sets the next goal as active.

Reply:
@MartensCedric yeah yeah before lockdown I was gonna ask some of the technical guys what the actual value should be since I really have no way of knowing that, like idk if 1.0 means 1 meter or 1 cm in rover terms, @vashmata could probably think of a realistic number so it should not be 50 (it just made it easier to test when it was a bit larger)

Reply:
if this is about coordinates then it's latitude and longitude haha, pretty sure it's in degrees

Reply:
for URC anyway, which is what this is designed for

Reply:
It's actually rover to destination distance
@vashmata. And if it uses the Nav_funs.py functions, then the distance is most likely in meters.